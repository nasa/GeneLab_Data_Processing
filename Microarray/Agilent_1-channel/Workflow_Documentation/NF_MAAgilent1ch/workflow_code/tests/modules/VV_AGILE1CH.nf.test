nextflow_process {

    name "Test Process VV_AGILE1CH"
    script "modules/VV_AGILE1CH.nf"
    process "VV_AGILE1CH"

    test("GLDS-367: Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                gldsAccession = "GLDS-367"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("test-datasets-extended/testdata/GLDS-367/Metadata/*")
                input[1] = file("test-datasets-extended/testdata/GLDS-367/00-DE/*")
                input[2] = false // val(skipVV)
                input[3] = file("${baseDir}/bin/dp_tools__agilent_1_channel")
                """
            }
        }

        then {
            assert process.success
            // TODO: future on dp_tools, messages in VV log are not deterministic due to use of sets to present items (i.e. order is not guaranteed)
            assert snapshot(
                process.out.VVed_raw_reads,
                process.out.VVed_runsheet,
                process.out.versions
                // process.out.log.get(0).lines(), // returns "parallel": false
                // process.out.log.get(0).lines().count(), // returns 1
                // process.out.log.get(0).count() // raises java.lang.NullPointerException
                // process.out.log.get(0) // gives file and md5sum default
                // process.out.log.get(0).readLines().count() // raises MethodSelectionException for list obj
                // process.out.log.get(0).readLines().size() // returns 1
                // ["V&V Log lines count" : process.out.log.get(0).countLines()] // Raises countLines error for get(0), seems to be a null pointer?
            ).match()
        }

    }
    test("GLDS-542: Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                gldsAccession = "GLDS-367"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("test-datasets-extended/testdata/GLDS-367/Metadata/*")
                input[1] = file("test-datasets-extended/testdata/GLDS-367/00-DE/*")
                input[2] = false // val(skipVV)
                input[3] = file("${baseDir}/bin/dp_tools__agilent_1_channel")
                """
            }
        }

        then {
            assert process.success
            // TODO: future on dp_tools, messages in VV log are not deterministic due to use of sets to present items (i.e. order is not guaranteed)
            assert snapshot(
                process.out.VVed_raw_reads,
                process.out.VVed_runsheet,
                process.out.versions
            ).match()
        }

    }

}
