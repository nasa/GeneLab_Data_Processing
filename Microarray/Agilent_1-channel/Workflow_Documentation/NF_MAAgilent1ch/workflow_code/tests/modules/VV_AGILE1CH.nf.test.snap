{
    "GLDS-542: Should run without failures": {
        "content": [
            [
                [
                    "SampleTable.csv:md5,f393473808e2b9e262c7da5809e0baca",
                    "contrasts.csv:md5,c9af9056cef7bd0210a626cd9c6175b2",
                    "differential_expression.csv:md5,1bce5936134e25585716ebe983743c32",
                    "normalized_expression.csv:md5,baae31c588f25cf574df54ee4937cbaa",
                    "raw_intensities.csv:md5,21b8fe8485e30baabd33d4341be08a78",
                    "visualization_PCA_table.csv:md5,6b37103ca8cacdfc2a924631e7422fa9",
                    "visualization_output_table.csv:md5,e1aa48627dde500d26443891de6c23d9"
                ]
            ],
            [
                "OSD-367_microarray_v0_runsheet.csv:md5,20d3407f692abaf4471151a7d85c0420"
            ],
            [
                "versions.yml:md5,a6554ecf93aa1f25709a8cce92f6c64e"
            ]
        ],
        "timestamp": "2023-02-23T22:33:56+0000"
    },
    "GLDS-367: Should run without failures": {
        "content": [
            [
                [
                    "SampleTable.csv:md5,f393473808e2b9e262c7da5809e0baca",
                    "contrasts.csv:md5,c9af9056cef7bd0210a626cd9c6175b2",
                    "differential_expression.csv:md5,1bce5936134e25585716ebe983743c32",
                    "normalized_expression.csv:md5,baae31c588f25cf574df54ee4937cbaa",
                    "raw_intensities.csv:md5,21b8fe8485e30baabd33d4341be08a78",
                    "visualization_PCA_table.csv:md5,6b37103ca8cacdfc2a924631e7422fa9",
                    "visualization_output_table.csv:md5,e1aa48627dde500d26443891de6c23d9"
                ]
            ],
            [
                "OSD-367_microarray_v0_runsheet.csv:md5,20d3407f692abaf4471151a7d85c0420"
            ],
            [
                "versions.yml:md5,a6554ecf93aa1f25709a8cce92f6c64e"
            ]
        ],
        "timestamp": "2023-02-23T22:33:55+0000"
    },
    "Should run without failures": {
        "content": [
            [
                [
                    "SampleTable.csv:md5,f393473808e2b9e262c7da5809e0baca",
                    "contrasts.csv:md5,c9af9056cef7bd0210a626cd9c6175b2",
                    "differential_expression.csv:md5,7aabea95a4b1d09ec26fb826daa17f08",
                    "visualization_PCA_table.csv:md5,6b37103ca8cacdfc2a924631e7422fa9",
                    "visualization_output_table.csv:md5,6cac37993daf664d1be489b1f5a9befd"
                ]
            ],
            [
                "GLDS-367_microarray_v0_runsheet.csv:md5,6b066905180d502a70e4ad9fc094433d"
            ]
        ],
        "timestamp": "2023-02-23T22:33:55+0000"
    }
}